File: making_decisions/python/multistate_sales_tax_calculator.py
# Pratice 20. Multistate sales tax calculator
# Output:
#   What is the order amount? 10
#   What state do you live in? Wisconsin
#   What county do you live in? Eau Claire
#   The state tax is $0.55.
#   The county tax is $0.05.
#   The total tax is $0.60.
#   The total is $10.60.
#   Or
#   What is the order amount? 10
#   What state do you live in? Illinois
#   The total tax is $0.80.
#   The total is $10.80.
# Standard:
#   State
#   Illinois : 0.08
#   County (If state is Wisconsin)
#       - Eau Claire : 0.005
#       - Dunn : 0.004
# Constraint:
#   - Ensure that all money is rounded up to the nearest cent.
#   - Use a single output statement at the end of the program to display the program results.

#!/usr/bin/env python

import sys

def input_process(in_question):
    return input(in_question) if sys.version_info >= (3,0) else raw_input(in_question)


if __name__ == '__main__':
    try:
        amount = int(input_process('What is the order amount? '))
        state = str(input_process('What state do you live in? '))

        exists_county = False
        if state.upper() == 'WISCONSIN':
            county = str(input_process('What county do you live in? '))
            exists_county = True
    except:
        print ('You must input only numbers.')
    else:
        county_tax = 0

        if state.upper() == 'WISCONSIN':
            if county.replace(' ','').upper() == 'EAUCLAIRE':
                county_tax = amount * 0.005
            elif county.replace(' ','').upper() == 'DUNN':
                county_tax = amount * 0.004
            else:
                county_tax = 0
            tax = amount * 0.055
        elif state.upper() == 'ILLINOIS':
            tax = amount * 0.08
        else:
            tax = 0

        print_str = 'The state tax is $%s\n' % round(tax, 2)

        if exists_county == True:
            print_str += 'The county tax is $%s\n' % round(county_tax, 2)

        print_str += 'The total tax is $%s\n' % round((tax + county_tax), 2)
        print_str += 'The total is $%s' % round((amount + tax + county_tax), 2)

        print (print_str)

File: making_decisions/go/multistate_sales_tax_calculator.go
/*
 * Pratice 20. Multistate sales tax calculator
 * Output:
 *   What is the order amount? 10
 *   What state do you live in? Wisconsin
 *   What county do you live in? Eau Claire
 *   The state tax is $0.55.
 *   The county tax is $0.05.
 *   The total tax is $0.60.
 *   The total is $10.60.
 *   Or
 *   What is the order amount? 10
 *   What state do you live in? Illinois
 *   The total tax is $0.80.
 *   The total is $10.80.
 * Standard:
 *   State
 *   Illinois : 0.08
 *   County (If state is Wisconsin)
 *       - Eau Claire : 0.005
 *       - Dunn : 0.004
 * Constraint:
 *   - Ensure that all money is rounded up to the nearest cent.
 *   - Use a single output statement at the end of the program to display the program results.
 */

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("What is the order amount? ")
	amount_str, _ := reader.ReadString('\n')

	fmt.Println("What state do you live in? ")
	state, _ := reader.ReadString('\n')

	amount, _ := strconv.ParseFloat(strings.TrimSpace(amount_str), 64)

	tax := 0.0
	county_tax := 0.0

	if strings.ToUpper(strings.TrimSpace(state)) == "WISCONSIN" {
		fmt.Println("What county do you live in? ")
		county, _ := reader.ReadString('\n')

		if strings.ToUpper(strings.TrimSpace(county)) == "EAU CLAIRE" {
			county_tax = amount * 0.005
		} else if strings.ToUpper(strings.TrimSpace(county)) == "DUMN" {
			county_tax = amount * 0.004
		}

		tax = amount * 0.055
	} else if strings.ToUpper(strings.TrimSpace(state)) == "ILLINOIS" {
		tax = amount * 0.08
	} else {
		tax = 0
	}

	fmt.Printf("The state tax is $%.2f\n", tax)

	if county_tax != 0 {
		fmt.Printf("The county tax is $%.2f\n", county_tax)
	}

	fmt.Printf("The total tax is $%.2f\n", tax+county_tax)
	fmt.Printf("The total is $%.2f\n", amount+tax+county_tax)
}
